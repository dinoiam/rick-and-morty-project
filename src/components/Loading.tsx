import React from 'react';
import Morty from '@root/images/morty.png';
import styled, { keyframes } from 'styled-components';

const dots = keyframes`
  0%, 20% {
    color: rgba(0,0,0,0);
    text-shadow:
      .25em 0 0 rgba(0,0,0,0),
      .5em 0 0 rgba(0,0,0,0);
  }
  40% {
    color: white;
    text-shadow:
      .25em 0 0 rgba(0,0,0,0),
      .5em 0 0 rgba(0,0,0,0);
  }
  60% {
    text-shadow:
      .25em 0 0 white,
      .5em 0 0 rgba(0,0,0,0);
  }
  80%, 100% {
    text-shadow:
      .25em 0 0 white,
      .5em 0 0 white;
  }
`;

const LoadingComponent = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;
const LoadingImage = styled.img`
  width: 70px;
  margin-bottom: 10px;
`;

const LoadingText = styled.p`
  color: rgb(158, 158, 158);

  &::after {
    content: ' .';
    animation: ${dots} 1s steps(5, end) infinite;
  }
`;
export default function Loading(): JSX.Element {
  return (
    <LoadingComponent>
      <LoadingImage alt="loading" src={Morty} />
      <LoadingText>Loading</LoadingText>
    </LoadingComponent>
  );
}
