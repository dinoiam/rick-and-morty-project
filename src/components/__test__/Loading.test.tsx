import React from 'react';
import Loading from '@components/Loading';
import { render } from '@testing-library/react';

describe('Loading', () => {
  test('it renders properly', () => {
    const { container } = render(<Loading></Loading>);
    expect(container).toMatchSnapshot();
  });
});
