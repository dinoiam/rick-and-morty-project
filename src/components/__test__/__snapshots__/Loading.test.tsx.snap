// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading it renders properly 1`] = `
<div>
  <div
    class="sc-bdfBwQ hmCjDE"
    data-testid="loading-component"
  >
    <img
      alt="loading"
      class="sc-gsTCUz eSbSna"
      src="div"
    />
    <p
      class="sc-dlfnbm hrhGhu"
    >
      Loading
    </p>
  </div>
</div>
`;
